docs/spring.md

```diff
@@ -38,7 +40,147 @@ details).
 
 ## Examples
 
-Check out the samples:
+### Spring MVC
+
+There is a `CloudEventHttpMessageConverter` that you can register for Spring MVC:
+
+```java
+@Configuration
+public static class CloudEventHandlerConfiguration implements WebMvcConfigurer {
+
+    @Override
+    public void configureMessageConverters(List<HttpMessageConverter<?>> converters) {
+        converters.add(0, new CloudEventHttpMessageConverter());
+    }
+
+}
+```
+
+With this in place you can write a `@RestController` with `CloudEvent` inputs or outputs, and the conversion will be handled by Spring. Example "echo" endpoint:
+
+```java
+@PostMapping("/echo")
+public CloudEvent ce(@RequestBody CloudEvent event) {
+    return CloudEventBuilder.from(event) //
+            .withId(UUID.randomUUID().toString()) //
+            .withSource(URI.create("https://spring.io/foos")) //
+            .withType("io.spring.event.Foo") //
+            .withData(event.getData().toBytes()) //
+            .build();
+}
+```
+
+Both structured and unstructured events are supported. So if you know that the `CloudEvent` data can be converted to a `Foo` then you can also use the `CloudEventHttpUtils` to deal with HTTP headers and stick to POJOs in the handler method. Example:
```

Comment:
s/unstructured/binary

Reply:
> So if you know that the `CloudEvent` data can be converted to a `Foo`

More than this, I think you should explain that you can do that only if you're certain the input is a binary cloudevent. So maybe what about:

> So if you know that the input `CloudEvent` is in HTTP binary mode and the data can be converted to a `Foo`